src/main/java/core/basesyntax/operations/ReturnOperation.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(Storage storage, FruitTransaction transaction) {
+        if (storage == null) {
+            throw new IllegalArgumentException("Storage cannot be null.");
+        }
+        if (transaction == null) {
+            throw new IllegalArgumentException("Transaction cannot be null.");
+        }
+        if (transaction.getQuantity() <= 0) {
+            throw new IllegalArgumentException("Quantity must be greater than zero.");
+        }
+
+        int currentQuantity = storage.getQuantity(transaction.getFruit());
+        int newQuantity = currentQuantity + transaction.getQuantity();
+
+        storage.updateFruit(transaction.getFruit(), newQuantity);

```

Comment:
The `apply` method in the `ReturnOperation` class is currently handling multiple responsibilities: validating the `storage` and `transaction`, as well as performing the return operation. Consider refactoring this method to follow the Single Responsibility Principle (SRP), one of the SOLID principles. You could extract the validation logic into separate private methods to make the `apply` method more focused on the return operation itself.